src/frontend/src/optimizer/rule/stream/filter_with_now_to_join_rule.rs

```diff
@@ -55,11 +54,6 @@ impl Rule for FilterWithNowToJoinRule {
             }
         });
 
-        // We want to put `input_expr >/>= now_expr` before `input_expr </<= now_expr` as the former
-        // will introduce a watermark that can reduce state (since `now_expr` is monotonically
-        // increasing)
-        now_filters.sort_by_key(|l| rank_cmp(l.func_type()));
```

Comment:
After https://github.com/risingwavelabs/risingwave/pull/13985, we can do state cleaning for the temporal filter in any direct, So the order of the temporal filters is not important.